### PR TITLE
[TINKERPOP-3016] Fix for reading value from different read operation

### PR DESCRIPTION
**Pull request by vkagamlyk**
Fix for issue when `TinkerTransactionGraph` used without transaction. A situation is possible when the value read before by another client in the same thread is returned.

Something like
Client1 in Thread1 read Vertex1, value cashed for consistent read.
Tx created for this request commit all changes, but for read operation it's nothing to do.

Client2 in Thread2 modify Vertex1 and commit changes.

Client3 got same Thread1 for his request, but cashed Vertex1 still used.

https://issues.apache.org/jira/browse/TINKERPOP-3016

### CODE REVIEW

**Comment by kenhuuu** [comment]
VOTE +1

**Comment by kenhuuu** [comment]
Needs a CHANGELOG.

File: tinkergraph-gremlin/src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/structure/TinkerTransactionGraph.java [hunk 1]
```diff
@@ -72,8 +72,8 @@ public final class TinkerTransactionGraph extends AbstractTinkerGraph {
 
     private final TinkerTransaction transaction = new TinkerTransaction(this);
 
-    protected Map<Object, TinkerElementContainer<TinkerVertex>> vertices = new ConcurrentHashMap<>();
-    protected Map<Object, TinkerElementContainer<TinkerEdge>> edges = new ConcurrentHashMap<>();
+    final Map<Object, TinkerElementContainer<TinkerVertex>> vertices = new ConcurrentHashMap<>();
```
Comment (kenhuuu) [inline tinkergraph-gremlin/src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/structure/TinkerTransactionGraph.java:75]: Nit: explicitly state what the access modifer should be.

File: tinkergraph-gremlin/src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/structure/TinkerElementContainer.java [hunk 2]
```diff
@@ -221,10 +235,11 @@ public boolean canBeRemoved() {
     /**
      * Cleanup changes made in the current transaction.
      */
-    private void reset() {
+    void reset() {
```
Comment (kenhuuu) [inline tinkergraph-gremlin/src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/structure/TinkerElementContainer.java:238]: Should this be public?

**Comment by Cole-Greer** [comment]
VOTE +1

**Comment by vkagamlyk** [comment]
VOTE+1

**Pull request merged**